cavern/src/main/java/org/opencadc/cavern/files/FileAction.java

```diff
@@ -70,51 +70,80 @@
 import ca.nrc.cadc.rest.InlineContentHandler;
 import ca.nrc.cadc.rest.RestAction;
 import ca.nrc.cadc.util.PropertiesReader;
+
+import ca.nrc.cadc.util.StringUtil;
+import ca.nrc.cadc.vos.ContainerNode;
+import ca.nrc.cadc.vos.DataNode;
 import ca.nrc.cadc.vos.Direction;
+import ca.nrc.cadc.vos.LinkingException;
+import ca.nrc.cadc.vos.Node;
+import ca.nrc.cadc.vos.NodeLockedException;
+import ca.nrc.cadc.vos.NodeNotFoundException;
+import ca.nrc.cadc.vos.NodeNotSupportedException;
 import ca.nrc.cadc.vos.VOSURI;
 
+import ca.nrc.cadc.vos.server.LocalServiceURI;
+import ca.nrc.cadc.vos.server.PathResolver;
+import ca.nrc.cadc.vos.server.auth.VOSpaceAuthorizer;
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.nio.file.attribute.GroupPrincipal;
 import java.nio.file.attribute.UserPrincipalLookupService;
 import java.security.AccessControlException;
 
 import org.apache.log4j.Logger;
-import org.opencadc.cavern.PosixIdentityManager;
+import org.opencadc.cavern.FileSystemNodePersistence;
 
 /**
  *
  * @author majorb
+ * @author jeevesh
  */
 public abstract class FileAction extends RestAction {
     private static final Logger log = Logger.getLogger(FileAction.class);
 
+    // Key values needed for FileAction
+    private VOSURI nodeURI;
+    private boolean isPreauth;
+
+    // Supporting tools and values
     private String root;
     private UserPrincipalLookupService upLookupSvc;
-    private PosixIdentityManager identityManager;
+    private VOSpaceAuthorizer authorizer;
+    protected PathResolver pathResolver;
+    private FileSystemNodePersistence fsPersistence;
 
-    private VOSURI nodeURI;
+    protected FileAction(boolean isPreauth) {
+        this.isPreauth = isPreauth;
 
-    protected FileAction() {
+        // Set up tools needed for generating nodeURI and
+        // validating permissions
         PropertiesReader pr = new PropertiesReader("Cavern.properties");
         this.root = pr.getFirstPropertyValue("VOS_FILESYSTEM_ROOT");
-        if (this.root == null) {
+        if (root == null) {
```

Comment:
This is nit-picky, but the references to `root` should be consistent with the `this.root` notation.

Reply:
ok.